### DISCUSSION TITLE
0x602 Network Error Ark is broke

### DISCUSSION THREAD

**Discussion by Redstoneguy129** [GameServerManagers/LinuxGSM-Docker#59] [Q&A] [answered]
`Update state (0x11) preallocating, progress: 99.76 (17963936604 / 18006952591)
 Update state (0x11) preallocating, progress: 99.76 (17964478131 / 18006952591)
 Update state (0x11) preallocating, progress: 99.76 (17964525029 / 18006952591)
 Update state (0x11) preallocating, progress: 99.77 (17964668355 / 18006952591)
 Update state (0x11) preallocating, progress: 99.77 (17965413927 / 18006952591)
 Update state (0x201) stopping, progress: 0.00 (0 / 0)
Error! App '376030' state is 0x602 after update job.
Apr 21 19:30:37.502 arkserver: INSTALL: ERROR: Installing arkserver: SteamCMD: Update required but not completed - check network
Redirecting stderr to '/home/linuxgsm/.local/share/Steam/logs/stderr.txt'
[  0%] Checking for available updates...
[----] Verifying installation...
Steam Console Client (c) Valve Corporation - version 1648077083
-- type 'quit' to exit --
Loading Steam API...OK

Connecting anonymously to Steam Public...OK
Waiting for client config...OK
Waiting for user info...OK
 Update state (0x3) reconfiguring, progress: 0.00 (0 / 0)
 Update state (0x11) preallocating, progress: 0.00 (0 / 18006952591)
 Update state (0x11) preallocating, progress: 1.31 (235266364 / 18006952591)
 Update state (0x11) preallocating, progress: 2.00 (360664529 / 18006952591)`
 
 As you can see i am following the install process then at the end of it the install fails and spits out a network error. I have run this 12 times. I am also using the default configuration. So either the latest ark is broke, the steam server is cutting out or you have shipped a buggy image. Everywhere online i see others have encountered same error except they didn't wait 2 hours of installing ark to encounter it. And no there are no permission problems.
 
------- Update 21/04/2022 23:00 ---------
Looked into steam log files and found only these errors
`[2022-04-21 19:30:29] Failed to write chunk in file "linux64\steamclient.so", 1048576 bytes at offset 0 (Write Failed)
[2022-04-21 19:30:29] AppID 376030 update canceled : Failed updating depot 1006 while writing chunk, offset 0 (Write Failed) (Disk write failure) "/home/linuxgsm/serverfiles/steamapps/downloading/376030/linux64/steamclient.so"`

Just going to add i don't have a full disk, i have 94TB free with a total of 155TB.
My internet speed is 190MB/s download with a fast business connection so i shouldn't be getting dropped out.
I have correct permissions set for the linuxgsm user so there shouldn't be a write or read file issue either.

**Comment by Claiyc** [comment]
Not an issue with lgsm, converting to discussion.

**Reply by Redstoneguy129** [reply]
Does anyone know the reason it breaks at the end and not immediately like the other scenarios with this error?

**Comment by jusito** [comment]
Tested with my pr, installed arkserver in 5-6 minutes. So at least its not on ark/steam side, and like already noted doesn't look like an lgsm issue. lgsm is just invoking steamcmd and it fails during this invocation.
Reboot month ago without ecc ram? SMART values are fine? Memtest is fine for _every_ ram?
All of this happened to me on my servers, so that is my usual checklist.

**Comment by Redstoneguy129** [comment, marked as answer]
I found the problem, and it's todo with how docker manages the disk.
The volume that stores my docker daemon is only 30GB big and i usually use a external cifs volume mount for my containers. For some reason docker has trouble working with permissions on cifs mounts and so even though looking at the permissions on the client, it's a different story for the server. Essentially the container was being told it has the right permissions but the way docker handled the cifs mount meant it didn't really have the permissions the container needed.
This was proven when i decided to instead run the container on my windows docker daemon as it is built differently to the normal docker daemon to work with windows mount types as cifs is used for samba shares for windows computers normally.
The container downloaded ARK fine and within 20 minutes on windows so i'm going to look into different mounting methods for my docker volume. So this problem wasn't anything to do with how linuxgsm or steamcmd. It was to do with how docker manages drives.
